### PR TITLE
Move custom shaders guide into the `Documentation` directory

### PR DESCRIPTION
**Pull request by ptrgags**
Custom Shaders does have documentation, it just was hiding in the `ModelExperimental` folder while under development. I'm moving it to a more discoverable place, in `Documentation/Contributors/CustomShaderGuide/README.md`. 

@ggetz I'm not 100% certain if this is the right subfolder. If there's a better place to put it, please let me know and I'll update this PR.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ptrgags!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/move-custom-shaders-guide/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

**Comment by ggetz** [comment]
I agree that its a bit out of place to put this in `Documentation/Contributors`. To be consistent with the only other non-contributor documentation-- the fabric schema`-- I think this should live in  `Documentation` directly, so maybe `Documentation/CustomShaderGuide/README.md`?

File: Documentation/Contributors/CustomShaderGuide/README.md [hunk 1]
```diff
@@ -0,0 +1,559 @@
+# `CustomShader` Documentation
+
+**Note**: This README is stored in `ModelExperimental/` temporarily while
```
Comment (ggetz) [inline Documentation/Contributors/CustomShaderGuide/README.md:3]: This note is no longer needed, correct?

Comment (ptrgags) [reply Documentation/Contributors/CustomShaderGuide/README.md:3]: Ah correct, I'll update this

**Comment by ptrgags** [comment]
@ggetz updated! I also added a changelog entry

**Comment by ptrgags** [comment]
@ggetz yes, this guide was reviewed when Custom Shaders was originally added, and again recently in #10018

**Pull request merged**